pkg/proc/fncall.go

```diff
@@ -391,48 +383,69 @@ func callOP(bi *BinaryInfo, thread Thread, regs Registers, callAddr uint64) erro
 	return thread.SetPC(callAddr)
 }
 
-// funcCallEvalExpr evaluates expr, which must be a function call, returns
-// the function being called and its arguments.
-func (scope *EvalScope) funcCallEvalExpr(callexpr *ast.CallExpr) (fn *Function, closureAddr uint64, argvars []*Variable, err error) {
+// funcCallEvalFuncExpr evaluates expr.Fun and returns the function that we're trying to call.
+// If allowCalls is false function calls will be disabled even if scope.callCtx != nil
+func funcCallEvalFuncExpr(scope *EvalScope, fncall *functionCallState, allowCalls bool) error {
 	bi := scope.BinInfo
 
-	fnvar, err := scope.evalAST(callexpr.Fun)
-	if err != nil {
-		return nil, 0, nil, err
+	if !allowCalls {
```

Comment:
Why not just bail here instead of continuing if function calls are disabled?

Reply:
It can still succeed and it will in most cases. It will only fail for something like `f()()` where the left side of a function call expression requires itself a nested function call, for those we have to evaluate the nested call (to `f()`) after we started the outer call (to the return value of `f`) so that we have space on the stack to save the return value of `f`.